src/pages/EnvironmentsPage/environmentsPermissions.ts

```diff
@@ -1,28 +1,35 @@
-import { LeoResourcePermissionsProvider } from 'src/analysis/Environments/Environments.models';
-import { getCreatorForCompute } from 'src/analysis/utils/resource-utils';
-import { App } from 'src/libs/ajax/leonardo/models/app-models';
+import { DecoratedComputeResource, PermissionsAndStateProvider } from 'src/analysis/Environments/Environments.models';
+import { getCreatorForCompute, isResourceDeletable } from 'src/analysis/utils/resource-utils';
+import { cromwellAppToolLabels } from 'src/analysis/utils/tool-utils';
+import { App, isApp } from 'src/libs/ajax/leonardo/models/app-models';
 import { PersistentDisk } from 'src/libs/ajax/leonardo/models/disk-models';
-import { ListRuntimeItem } from 'src/libs/ajax/leonardo/models/runtime-models';
+import { ListRuntimeItem, Runtime } from 'src/libs/ajax/leonardo/models/runtime-models';
 import { getTerraUser } from 'src/libs/state';
 
-const makePermissionsProvider = (userEmailGetter: () => string): LeoResourcePermissionsProvider => {
-  const permissionsProvider: LeoResourcePermissionsProvider = {
+const makeStateAndPermissionsProvider = (userEmailGetter: () => string): PermissionsAndStateProvider => {
+  return <PermissionsAndStateProvider>{
     canDeleteDisk: (disk: PersistentDisk) => {
       const currentUserEmail = userEmailGetter();
-      const canDelete = disk.auditInfo.creator === currentUserEmail;
-      return canDelete;
+      return disk.auditInfo.creator === currentUserEmail;
     },
     canPauseResource: (resource: App | ListRuntimeItem) => {
       const currentUserEmail = userEmailGetter();
       const creator = getCreatorForCompute(resource);
       return currentUserEmail === creator;
     },
+    canDeleteApp: (resource: DecoratedComputeResource) => {
+      if (isApp(resource)) {
+        return !Object.keys(cromwellAppToolLabels).includes(resource.appType);
```

Comment:
Does this only filter our CROMWELL apps? What about WORKFLOWS or WDS apps? Are they also un-deleteable?

And should we include something like the pause logic which checks whether the deleting user matches the creator?

Reply:
This only filters out our CROMWELL apps, so `WORKFLOWS_APP` `CROMWELL_RUNNER_APP` and `CROMWELL` app types. Should WDS be included as well?

I think adding that logic would make sense, but only after we allow users to actually delete their own apps. I think once we enable that function, we should also make sure that the person deleting the app should have permission to do so.